components/expenses/ExpenseFormPayeeStep.js

```diff
@@ -110,7 +112,13 @@ const getPayeeOptions = (intl, payoutProfiles) => {
     [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true,
   }));
 
-  const [myself, myOrganizations] = partition(profileOptions, p => p.value.type === 'INDIVIDUAL');
+  const myself = profileOptions.filter(p => p.value.type === INDIVIDUAL);
+  const myOrganizations = profileOptions.filter(p => p.value.type === ORGANIZATION);
+  const myCollectives = profileOptions.filter(p => p.value.type === COLLECTIVE);
+  const myFunds = profileOptions.filter(p => p.value.type === FUND);
+  const myProjects = profileOptions.filter(p => p.value.type === PROJECT);
+  const myEvents = profileOptions.filter(p => p.value.type === EVENT);
```

Comment:
:ice_cream: Personal pref: a `groupBy` seems simpler to me than iterating 6 times on the list. This won't have any significant impact on performance since this function is memoized.

```es6
const profilesByType = groupBy(profileOptions, p => p.value.type);

if (profilesByType[COLLECTIVE]?.length) {
  payeeOptions.push({ options: profilesByType[COLLECTIVE], label: intl.formatMessage({ defaultMessage: 'My Collectives' }) });
}
```